tests/pruners_tests/test_median.py

```diff
@@ -9,97 +9,97 @@
 
 def test_median_pruner_with_one_trial() -> None:
 
-    study = optuna.study.create_study()
-    trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
-    trial.report(1, 1)
     pruner = optuna.pruners.MedianPruner(0, 0)
+    study = optuna.study.create_study(pruner=pruner)
+    trial = study.ask()
+    trial.report(1, 1)
 
     # A pruner is not activated at a first trial.
-    assert not pruner.prune(study=study, trial=study._storage.get_trial(trial._trial_id))
+    assert not trial.should_prune()
 
 
 @pytest.mark.parametrize("direction_value", [("minimize", 2), ("maximize", 0.5)])
 def test_median_pruner_intermediate_values(direction_value: Tuple[str, float]) -> None:
 
     direction, intermediate_value = direction_value
     pruner = optuna.pruners.MedianPruner(0, 0)
-    study = optuna.study.create_study(direction=direction)
+    study = optuna.study.create_study(direction=direction, pruner=pruner)
 
-    trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
+    trial = study.ask()
     trial.report(1, 1)
-    study._storage.set_trial_state(trial._trial_id, TrialState.COMPLETE)
+    study.tell(trial, 1)
 
-    trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
+    trial = study.ask()
     # A pruner is not activated if a trial has no intermediate values.
-    assert not pruner.prune(study=study, trial=study._storage.get_trial(trial._trial_id))
+    assert not trial.should_prune()
 
     trial.report(intermediate_value, 1)
     # A pruner is activated if a trial has an intermediate value.
-    assert pruner.prune(study=study, trial=study._storage.get_trial(trial._trial_id))
+    assert trial.should_prune()
 
 
 def test_median_pruner_intermediate_values_nan() -> None:
 
     pruner = optuna.pruners.MedianPruner(0, 0)
-    study = optuna.study.create_study()
+    study = optuna.study.create_study(pruner=pruner)
 
-    trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
+    trial = study.ask()
     trial.report(float("nan"), 1)
     # A pruner is not activated if the study does not have any previous trials.
-    assert not pruner.prune(study=study, trial=study._storage.get_trial(trial._trial_id))
-    study._storage.set_trial_state(trial._trial_id, TrialState.COMPLETE)
+    assert not trial.should_prune()
+    study.tell(trial, -1)
```

Comment:
What would be a good value here? `study.tell(trial, float("nan"))` rises ValueError.

Reply:
In my understanding, the value doesn't matter for the purpose of these unit tests so I don't have a strong opinion but something that communicates this, like -1 (or 0) sounds reasonable to me. We could also leave a comment.

Reply:
It also makes sense for me to use `-1` here.